src/main/java/picard/illumina/parser/readers/CbclReader.java

```diff
@@ -57,117 +63,178 @@
+                    }
 
-                short version = byteBuffer.getShort();
-                int headerSize = byteBuffer.getInt();
+                    final InputStream stream = open(bclFile, false, false, false);
+                    int read = stream.read(byteBuffer.array());
 
-                final ByteBuffer headerBuffer = ByteBuffer.allocate(headerSize - INITIAL_HEADER_SIZE);
-                headerBuffer.order(ByteOrder.LITTLE_ENDIAN);
+                    //we need to read the first 6 bytes to determine the header size
+                    if (read != INITIAL_HEADER_SIZE) {
+                        close();
+                        throw new RuntimeIOException(String.format("BCL %s has invalid header structure.", bclFile.getAbsoluteFile()));
+                    }
 
-                read = stream.read(headerBuffer.array());
-                if (read != headerSize - INITIAL_HEADER_SIZE) {
-                    close();
-                    throw new RuntimeIOException(String.format("BCL %s has invalid header structure.", bclFile.getAbsoluteFile()));
-                }
+                    short version = byteBuffer.getShort();
+                    int headerSize = byteBuffer.getInt();
 
-                byte bitsPerBasecall = headerBuffer.get();
-                byte bitsPerQualityScore = headerBuffer.get();
+                    final ByteBuffer headerBuffer = ByteBuffer.allocate(headerSize - INITIAL_HEADER_SIZE);
+                    headerBuffer.order(ByteOrder.LITTLE_ENDIAN);
 
-                if (bitsPerBasecall != 2 && bitsPerBasecall != bitsPerQualityScore) {
-                    close();
-                    throw new RuntimeIOException("CBCL data not encoded in nibbles. (not currently supported)");
-                }
+                    read = stream.read(headerBuffer.array());
+                    if (read != headerSize - INITIAL_HEADER_SIZE) {
+                        close();
+                        throw new PicardException(String.format("BCL %s has invalid header structure.", bclFile.getAbsoluteFile()));
+                    }
 
-                int numberOfBins = headerBuffer.getInt();
+                    byte bitsPerBasecall = headerBuffer.get();
+                    byte bitsPerQualityScore = headerBuffer.get();
 
-                byte[] qualityBins = new byte[numberOfBins];
-                //each bin has a pair of 4 byte mappings
-                for (int j = 0; j < numberOfBins; j++) {
-                    headerBuffer.getInt(); // first int is "from" value, which we don't need
-                    int to = headerBuffer.getInt();
-                    qualityBins[j] = (byte) to;
-                }
+                    if (bitsPerBasecall != 2 && bitsPerBasecall != bitsPerQualityScore) {
+                        close();
+                        throw new PicardException("CBCL data not encoded in nibbles. (not currently supported)");
```

Comment:
I was going to correct this as `nybble` but apparently nibble is the accepted spelling. Maybe this changed in the last 20 years?